js/main.js

```diff
@@ -331,8 +332,19 @@ function setCity(city) {
         updateLayers();
         updateUrlHash(city);
         document.title = 'Wo ist Markt in ' + cityIdToLabel(city) + '?';
+
         // Update drop down but avoid recursion
         $('#dropDownCitySelection').val(city).trigger('change', true);
+
+        if (firstTimeLoad) {
```

Comment:
The loading screen should also be shown during city change (see #30), so you don't need this check. Please refactor the code to show/hide the screen into separate functions that can then be called from `setCity` and `ready`.